### PR TITLE
Add _isLocked to AdaptModel

### PR DESCRIPTION
**Pull request by tomgreenfield**
Implementation as per #905.

### CODE REVIEW

**Comment by oliverfoster** [comment]
+1

**Comment by oliverfoster** [comment]
+1

**Comment by moloko** [comment]
this looks great @tomgreenfield - have you/@oliverfoster already given this a thorough testing?

File: src/core/js/models/adaptModel.js [hunk 1]
```diff
@@ -306,8 +308,80 @@ define(function (require) {
...
+                    this.setSequentialLocking();
+                    break;
+                case "unlockFirst":
+                    this.setUnlockFirstLocking();
+                    break;
+                case "lockLast":
+                    this.setLockLastLocking();
+                    break;
+                case "custom":
+                    this.setCustomLocking();
```
Comment (moloko) [inline src/core/js/models/adaptModel.js]: could you add a `default` case here that does something like:
`console.warn("AdaptModel.checkLocking: unknown _lockType of '" + this.get("_lockType") + "' found on " + this.get('_id'));`

**Comment by lc-thomasberger** [comment]
+1 (with default case and warning)

**Comment by brian-learningpool** [comment]
+1, I'm a little unsure of the use-case for `_isLocked` on articles, blocks and components though.

**Comment by hrajotia** [comment]
@tomgreenfield: `_isLocked` state of a model is incorrect on execution of `setupModel` method so can we call `this.checkLocking();` before [this.init()](https://github.com/tomgreenfield/adapt_framework/blob/issue/905/src/core/js/models/adaptModel.js#L43)?

**Comment by oliverfoster** [comment]
+1

**Comment by hrajotia** [comment]
+1

**Comment by chucklorenz** [comment]
+1

**Comment by moloko** [comment]
+1 tested all the possible locking types in combination with the latest boxmenu; all works exactly as designed. Great job @tomgreenfield!

**Pull request merged**